module/functions/azure/aad/Assert-AzureAdSecurityGroup.Tests.ps1

```diff
@@ -0,0 +1,198 @@
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+$sut = (Split-Path -Leaf $MyInvocation.MyCommand.Path).Replace(".Tests.ps1", ".ps1")
+
+. "$here\$sut"
+
+# define other functions that will be mocked
+function _EnsureAzureConnection {}
+function Get-AzADGroup {}
+function Invoke-AzRestMethod {}
+
+Describe "Assert-AzureAdSecurityGroup Tests" {
+
+    Mock _EnsureAzureConnection { $true }
+    Mock Write-Host {}
+
+    Context "Group does not exist" {
+
+        Mock Get-AzADGroup {}
+        Mock _buildUpdateRequest { @{Uri = 'https://fake'} }
+        Mock _getGroupOwners {}
+        Mock Invoke-AzRestMethod {}
+
+        Context "No default owners" {
+            $testGroup = @{
+                Name = 'testgroup'
+                EmailName = 'testgroup@nowhere.org'
+                Description = 'just a test group'
+                OwnersToAssignOnCreation = @()
+                StrictMode = $false
+            }
+            Assert-AzureAdSecurityGroup @testGroup
+
+            It "should create the group" {
+                Assert-MockCalled _buildUpdateRequest -Times 0
+                Assert-MockCalled _getGroupOwners -Times 0
+                Assert-MockCalled Invoke-AzRestMethod -Times 1
```

Comment:
This seems like a bit of a blunt instrument. Is there nothing we can do to verify that the REST method it invoked looks like it will do what we want?

Reply:
Good point, I've made some improvements to those test.